src/devices/Tlc1543/Tlc1543.cs

```diff
@@ -116,42 +135,196 @@ public int ReadChannel(int channelNumber)
+                        DelayHelper.Delay(_conversionTime, false);
+                    }
+
+                    break;
+                case Mode.Mode3:
+                    for (int i = 0; i < 16; i++)
+                    {
+                        if (i < 4)
+                        {
+                            SendAddress(channelNumber, i);
+                        }
+
+                        // read 10-bit data
+                        if (i <= 10)
+                        {
+                            _value <<= 1;
+                            _value |= (int)_controller.Read(_dataOut);
+                        }
+
+                        if (i > 10 && _controller.Read(_endOfConversion) == PinValue.High)
+                        {
+                            break;
+                        }
+
+                        ClockImpulse();
+                    }
+
+                    _controller.Write(_chipSelect, 1);
+                    break;
+                case Mode.Mode4:
+                    for (int i = 0; i < 16; i++)
+                    {
+                        if (i < 4)
+                        {
+                            SendAddress(channelNumber, i);
+                        }
+
+                        // read 10-bit data
+                        if (i <= 10)
+                        {
+                            _value <<= 1;
+                            _value |= (int)_controller.Read(_dataOut);
+                        }
+
+                        if (i > 10 && _controller.Read(_endOfConversion) == PinValue.High)
+                        {
+                            break;
+                        }
+
+                        ClockImpulse();
+                    }
+
+                    _controller.WaitForEvent(_endOfConversion, PinEventTypes.Rising, _conversionTime);
+
+                    break;
+                case Mode.Mode5:
+                    for (int i = 0; i < 16; i++)
+                    {
+                        if (i < 4)
+                        {
+                            SendAddress(channelNumber, i);
+                        }
+
+                        // read 10-bit data
+                        if (i <= 10)
+                        {
+                            _value <<= 1;
+                            _value |= (int)_controller.Read(_dataOut);
+                        }
+
+                        if (i > 11 && _controller.Read(_endOfConversion) == PinValue.High)
+                        {
+                            break;
+                        }
+
+                        ClockImpulse();
+                    }
 
-                _controller.Write(_inputOutputClock, 1);
-                DelayHelper.DelayMicroseconds(0, false);
-                _controller.Write(_inputOutputClock, 0);
+                    _controller.Write(_chipSelect, 1);
+                    break;
+                case Mode.Mode6:
+                    for (int i = 0; i < 16; i++)

```

Comment:
you're repeating exactly the same code in the first 10 cycles and this is re-implementing SPI even when you go to https://www.ti.com/product/TLC1543 it says "Interface type: SPI".